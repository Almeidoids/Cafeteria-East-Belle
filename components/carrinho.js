"use client"

import { useContext, useEffect, useState, createRef, useRef } from "react";

//Estilos
import "bootstrap/dist/css/bootstrap.min.css";
import styles from "../style/carrinho.module.css";
import "bootstrap-icons/font/bootstrap-icons.css";

//Constantes
import { Cart } from "../constants/carrinho";

//Funções
import { total } from "../functions/buy";

export default function Carrinho() {
    const {cart, setCart} = useContext(Cart);
    const cartLength = cart.length;
    const [listBuy, setListBuy] = useState([]);
    const liRefsDelete = useRef([]);
    const [save, setSave] = useState(false);
    const price = total(cart);

    useEffect(() => {
        if (localStorage.getItem("0") === null) {
            return;
        }
        else {
            setCart([]);
            const itemString = localStorage.getItem("0");
            const toRecover = JSON.parse(itemString);
            setCart(toRecover);
            setSave(JSON.parse(localStorage.getItem("1")));
        }
    }, [])

    useEffect(() => {
        if (cart.length === 0) return;
        else { 
            liRefsDelete.current = Array(cartLength).fill().map((_, i) => {
                liRefsDelete.current[i] || createRef();
            });
        }
    }, [cart]);

    useEffect(() => {
        let num = 0;
        setListBuy(cart.map(item => {
            return (
                <li 
                    style = {{display: "flex"}} 
                    key = {num++} 
                    onMouseEnter = {() => {liRefsDelete.current[cart.indexOf(item)].style.display = "block"}}
                    onMouseLeave = {() => {liRefsDelete.current[cart.indexOf(item)].style.display = "none"}} 
                >
                    <div className = {styles.lista}>
                        <p style = {{pointerEvents: "none"}}>{item.amount} {item.product}</p>
                        <p style = {{pointerEvents: "none"}}>{item.price}</p>
                    </div>
                    <button 
                        type = "button" 
                        className = {`btn-close`} 
                        style = {{marginLeft: 10, 
                        display: "none"}} 
                        aria-label = "Close"
                        ref = {e => liRefsDelete.current[cart.indexOf(item)] = e}
                        onClick = {() => {
                            setCart(cart.filter(value => value !== item))
                        }}
                    />
                </li>
            )
        }));
    }, [cart])

    useEffect(() => {
        if (save) {
            const toSave = JSON.stringify(cart);
            const saveToSave = JSON.stringify(save);
            localStorage.setItem("0", toSave);
            localStorage.setItem("1", saveToSave);
        }
        else {
            localStorage.removeItem("0");
            localStorage.setItem("1", JSON.stringify(save));
        }
    }, [listBuy, save])

    return (
            <div className = {`offcanvas offcanvas-end ${styles.carrinhoCorpo}`} data-bs-scroll = "true" data-bs-backdrop = "false" data-bs-theme = "dark" tabIndex = "-1" id = "offcanvasScrolling" aria-labelledby = "offcanvasScrollingLabel">
                <div className = "offcanvas-header">
                    <h5 className = "offcanvas-title" id = "offcanvasScrollingLabel" style = {{color: "#CAC1C7"}}>Carrinho</h5>
                    <button type = "button" className = {`btn-close`} style = {{marginLeft: "auto"}} data-bs-dismiss = "offcanvas" aria-label = "Close"/>
                </div>
                <div className = "offcanvas-body">
                    <ul className = {styles.ul}>
                        {listBuy}
                        {cart.length > 0 && 
                            <li className = {styles.lista}>
                                <p>Total</p>
                                <p>{price}</p>
                            </li>
                        }
                    </ul>
                    {cart.length > 0 &&
                        <div style = {{justifySelf: "flex-end"}}>
                            <button className = {`${styles.btnCancelar}`} onClick = {() => {
                                setCart([]);
                            }}>Cancelar</button>
                            <button type = "button" className = {`${styles.btnComprar}`} data-bs-toggle = "modal" data-bs-target = "#staticBackdrop">Comprar</button>
                        </div>
                    }
                    <label style = {{marginTop: 20}}>
                        <input type = "checkbox" className = {styles.checkbox} checked = {save} onChange = {(e) => setSave(e.target.checked)} /> Você deseja salvar o carrinho após sair do site?
                    </label>
                </div>
            </div>
    )
}
