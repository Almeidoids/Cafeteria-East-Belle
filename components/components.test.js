import { render, screen, fireEvent, waitFor, renderHook } from "@testing-library/react";
import Carrossel, * as carousel from "./carrossel";
import * as form from "./formulario";
import { Cart } from "../constants/carrinho";
import React from "react";
import CardItens from "./cardItens";
import Menu from "./menu";
import Receitas from "./receitas";
import Rodape from "./rodape";

describe("Carrossel", () => {
    const requestAnimationMock = jest.fn()
        
    beforeEach(() => {
        global.backup = global.requestAnimationFrame;
        global.requestAnimationFrame = requestAnimationMock;     
    })

    afterAll(() => {
        global.requestAnimationFrame = global.backup;
        delete global.backup;
    })

    afterEach(() => {
        jest.clearAllMocks();
    })

    test("Botão para girar o Carrossel", () => {
        
        const rotate = jest.spyOn(carousel.carousel, "carouselAnimation").mockImplementation(jest.fn());   
        render(<Carrossel><div style = {{width: "100%"}} /></Carrossel>);
        
        const buttonRight = screen.getByText(">");
        fireEvent.click(buttonRight);
        expect(carousel.carousel.carouselAnimation).toHaveBeenCalledWith(expect.any(Object), 0);
        const buttonLeft = screen.getByText("<");
        fireEvent.click(buttonLeft);
        expect(carousel.carousel.carouselAnimation).toHaveBeenCalledWith(expect.any(Object), 0);

        rotate.mockRestore();
    });

    test("Scroll do carrossel", () => {
        const carouselObject = {scrollLeft: 0, clientWidth: 500, scrollTo: jest.fn()};

        carousel.carousel.carouselAnimation(carouselObject, 250);
        let fn = requestAnimationMock.mock.calls[0][0];
        fn();
        fn();
        expect(carouselObject.scrollTo).toHaveBeenCalledTimes(2);
        expect(carouselObject.scrollTo.mock.calls[1][0]).toBe(500);

        carousel.carousel.carouselAnimation(carouselObject, -250);
        fn = requestAnimationMock.mock.calls[2][0];
        fn();
        fn();
        expect(carouselObject.scrollTo).toHaveBeenCalledTimes(4);
        console.log(carouselObject.scrollTo.mock.calls[3][0]);
        expect(carouselObject.scrollTo.mock.calls[3][0]).toBe(-500);
    })

    test("Componente Carrossel", () => {
        const test = render(<Carrossel><div /></Carrossel>);
        expect(test).toMatchSnapshot();
    });

    
})

describe("Formulario", () => {
    const product = {quantity: 10, key: 1, originalp: 10, offer: 5, name: "arroz"};
    const setCart = jest.fn(fn => fn([]));
    const setCartLength = jest.fn();

    test("Testando se o item é adicionado ao carrinho ao clicar no botão", async () => {

        render(
                <form onSubmit = {(e) => (form.addCart(e, product, setCart, setCartLength))}>
                    <input type = "number" defaultValue = {5} />
                    <button type = "submit">Enviar</button>
                </form>
        );
        
        const buttonSubmit = screen.getByText("Enviar");
        fireEvent.click(buttonSubmit);

        expect(setCart).toHaveBeenCalledTimes(1);
        expect(setCart).toHaveBeenCalledWith(expect.any(Function));
        expect(setCart).toHaveReturnedWith([{amount: 5, product: "arroz", price: "47.50", id: 1}]);
        expect(setCartLength).toHaveBeenCalledWith(1);
    })

    test("Testando se o input volta a ser 1 no final da chamada", () => {
        render(
            <form onSubmit = {(e) => (form.addCart(e, product, setCart, setCartLength))}>
                <input type = "number" defaultValue = {5} data-testid = "num" />
                <button type = "submit">Enviar</button>
            </form>
        );
        const inputNum = screen.getByTestId("num");
        expect(inputNum.value).toMatch("5");

        const buttonSubmit = screen.getByText("Enviar");
        fireEvent.click(buttonSubmit);
        expect(inputNum.value).toMatch("1");
    })
})

describe("outros componentes", () => {
    test("cardItens", () => {
        const component = render(
            <CardItens 
                key = {0}
                offer = {12}
                image = {"/images/affogato.png"}
                title = {"nome"}
                original = {"123"}
                link = {`/produtos/0`}
            />
        )

        expect(component).toMatchSnapshot();
    });

    test("Receitas", () => {
        const component = render(
            <Receitas close = {() => {return}} i = {0} />
        )

        expect(component).toMatchSnapshot();
    })

    test("Rodape", () => {
        const component = render(
            <Rodape />
        )

        expect(component).toMatchSnapshot();
    })


})