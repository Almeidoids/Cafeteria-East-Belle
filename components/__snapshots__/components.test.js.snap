// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrossel Componente Carrossel 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="organizacaoCarrossel"
      >
        <button
          class="botaoCarrossel"
        >
          &lt;
        </button>
        <div
          class="carrossel"
          data-testid="testeCarrossel"
        >
          <div />
        </div>
        <button
          class="botaoCarrossel"
        >
          &gt;
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="organizacaoCarrossel"
    >
      <button
        class="botaoCarrossel"
      >
        &lt;
      </button>
      <div
        class="carrossel"
        data-testid="testeCarrossel"
      >
        <div />
      </div>
      <button
        class="botaoCarrossel"
      >
        &gt;
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`outros componentes Receitas 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card corpo"
      >
        <div
          style="display: flex;"
        >
          <button
            aria-label="close"
            class="btn-close"
            style="margin-left: auto;"
            type="button"
          />
        </div>
        <h1
          style="text-align: center; margin-bottom: 20px;"
        >
          Café Etíope
        </h1>
        <div
          style="display: flex;"
        >
          <div
            class="conteudo"
          >
            <ol
              class="listaol"
            >
              <div>
                <li>
                  Ingredientes
                </li>
                <ul
                  class="ul"
                  style="font-family: fontFamily;"
                >
                  <li>
                    Grãos de café etiopes frescos
                  </li>
                  <li>
                    Água
                  </li>
                  <li>
                    Moedor de café
                  </li>
                  <li>
                    Filtro de papel ou pano (Dependendo do métodp)
                  </li>
                  <li>
                    Cafeteria de sua preferência
                  </li>
                  <li>
                    Açúcar (opcional)
                  </li>
                </ul>
              </div>
              <div>
                <li>
                  Preparo
                </li>
                <ol
                  class="subol"
                  style="font-family: fontFamily;"
                >
                  <li
                    class="lists"
                  >
                    <b>
                      Moa os grãos:
                    </b>
                     Moa os grãos de café de um moedor, ajustando a moagem de acordo com o método escolhido
                  </li>
                  <li
                    class="lists"
                  >
                    <b>
                      Preparando o café:
                    </b>
                  </li>
                  <ul
                    class="ul"
                    style="font-family: fontFamily;"
                  >
                    <li>
                      Usando filtro de papel, coloque-o na cafeteria e enxágue com água quente para remover o sabor do papel e pré-aquecer a cafeteira.
                    </li>
                    <li>
                      Aqueça a água até a temperatura ideal (90-95°C)
                    </li>
                    <li>
                      Coloque o café moido na cafeteira, seguindo as proporções recomendadas para o seu método de preparo.
                    </li>
                    <li>
                      Despeje a água quente sobre o café, garantindo que todos os grãos sejam saturados.
                    </li>
                    <li>
                      Aguarde o tempo de infusão necessário
                    </li>
                  </ul>
                  <li
                    class="lists"
                  >
                    <b>
                      Sirva o café:
                    </b>
                     Sirva o café quente e adicione açúcar se desejar.
                  </li>
                </ol>
              </div>
              <div
                style="margin-bottom: 10%;"
              >
                <li>
                  Dicas
                </li>
                <ul
                  class="ul"
                  style="font-family: fontFamily;"
                >
                  <li>
                    Use grãos de café etíopes de alta qualidade para um sabor autêntico+
                  </li>
                  <li>
                    Experimente diferentes métodos de preparo para descobrir o seu favorito.
                  </li>
                  <li>
                    Beba o café etíope com moderação, pois ele é forte e estimulante.
                  </li>
                </ul>
              </div>
            </ol>
          </div>
          <div
            class="divImagens"
          >
            <div>
              <img
                alt="Ingredientes Café Etiope"
                class="imagens"
                data-nimg="1"
                decoding="async"
                height="400"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=828&q=75"
                srcset="/_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=828&q=75 2x"
                style="color: transparent;"
                width="400"
              />
            </div>
            <div>
              <img
                alt="Preparo do Café Etiope"
                class="imagens"
                data-nimg="1"
                decoding="async"
                height="400"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=828&q=75"
                srcset="/_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=828&q=75 2x"
                style="color: transparent;"
                width="400"
              />
            </div>
            <div>
              <img
                alt="Mulher muito sorridente aproveitando de seu cafe"
                class="imagens"
                data-nimg="1"
                decoding="async"
                height="400"
                loading="lazy"
                src="/_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=828&q=75"
                srcset="/_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=828&q=75 2x"
                style="color: transparent;"
                width="400"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card corpo"
    >
      <div
        style="display: flex;"
      >
        <button
          aria-label="close"
          class="btn-close"
          style="margin-left: auto;"
          type="button"
        />
      </div>
      <h1
        style="text-align: center; margin-bottom: 20px;"
      >
        Café Etíope
      </h1>
      <div
        style="display: flex;"
      >
        <div
          class="conteudo"
        >
          <ol
            class="listaol"
          >
            <div>
              <li>
                Ingredientes
              </li>
              <ul
                class="ul"
                style="font-family: fontFamily;"
              >
                <li>
                  Grãos de café etiopes frescos
                </li>
                <li>
                  Água
                </li>
                <li>
                  Moedor de café
                </li>
                <li>
                  Filtro de papel ou pano (Dependendo do métodp)
                </li>
                <li>
                  Cafeteria de sua preferência
                </li>
                <li>
                  Açúcar (opcional)
                </li>
              </ul>
            </div>
            <div>
              <li>
                Preparo
              </li>
              <ol
                class="subol"
                style="font-family: fontFamily;"
              >
                <li
                  class="lists"
                >
                  <b>
                    Moa os grãos:
                  </b>
                   Moa os grãos de café de um moedor, ajustando a moagem de acordo com o método escolhido
                </li>
                <li
                  class="lists"
                >
                  <b>
                    Preparando o café:
                  </b>
                </li>
                <ul
                  class="ul"
                  style="font-family: fontFamily;"
                >
                  <li>
                    Usando filtro de papel, coloque-o na cafeteria e enxágue com água quente para remover o sabor do papel e pré-aquecer a cafeteira.
                  </li>
                  <li>
                    Aqueça a água até a temperatura ideal (90-95°C)
                  </li>
                  <li>
                    Coloque o café moido na cafeteira, seguindo as proporções recomendadas para o seu método de preparo.
                  </li>
                  <li>
                    Despeje a água quente sobre o café, garantindo que todos os grãos sejam saturados.
                  </li>
                  <li>
                    Aguarde o tempo de infusão necessário
                  </li>
                </ul>
                <li
                  class="lists"
                >
                  <b>
                    Sirva o café:
                  </b>
                   Sirva o café quente e adicione açúcar se desejar.
                </li>
              </ol>
            </div>
            <div
              style="margin-bottom: 10%;"
            >
              <li>
                Dicas
              </li>
              <ul
                class="ul"
                style="font-family: fontFamily;"
              >
                <li>
                  Use grãos de café etíopes de alta qualidade para um sabor autêntico+
                </li>
                <li>
                  Experimente diferentes métodos de preparo para descobrir o seu favorito.
                </li>
                <li>
                  Beba o café etíope com moderação, pois ele é forte e estimulante.
                </li>
              </ul>
            </div>
          </ol>
        </div>
        <div
          class="divImagens"
        >
          <div>
            <img
              alt="Ingredientes Café Etiope"
              class="imagens"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=828&q=75"
              srcset="/_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FingredientesEtiope.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div>
            <img
              alt="Preparo do Café Etiope"
              class="imagens"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=828&q=75"
              srcset="/_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FpreparoEtiope.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
          <div>
            <img
              alt="Mulher muito sorridente aproveitando de seu cafe"
              class="imagens"
              data-nimg="1"
              decoding="async"
              height="400"
              loading="lazy"
              src="/_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=828&q=75"
              srcset="/_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Freceitas%2FdicasEtiope.jpg&w=828&q=75 2x"
              style="color: transparent;"
              width="400"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`outros componentes Rodape 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="footer variable undefined"
      >
        <div
          class="column"
        >
          <h4
            class="h4"
          >
            Sobre nós
          </h4>
          <a
            href="/sobre"
          >
            <p
              class="textoFooter"
            >
              Nossa Cafeteria
            </p>
          </a>
          <a
            href="/sobre/inspiracao"
          >
            <p
              class="textoFooter"
            >
              Nossa Inspiração
            </p>
          </a>
          <a
            href="/sobre/meioAmbiente"
          >
            <p
              class="textoFooter"
            >
              Meio-Ambiente
            </p>
          </a>
          <a
            href="/sobre/contato"
          >
            <p
              class="textoFooter"
            >
              Contate-nos
            </p>
          </a>
        </div>
        <div
          class="column"
        >
          <h4
            class="h4"
          >
            Produtos
          </h4>
          <p
            class="textoFooter"
          >
            Cadastrar produtos
          </p>
          <p
            class="textoFooter"
          >
            Alterar Produtos
          </p>
          <p
            class="textoFooter"
          >
            Visualizar Produtos
          </p>
        </div>
        <div
          class="column"
        >
          <h4
            class="h4"
          >
            Usuarios
          </h4>
          <p
            class="textoFooter"
          >
            Cadastre-se
          </p>
          <p
            class="textoFooter"
          >
            Login
          </p>
          <p
            class="textoFooter"
          >
            Carrinho
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="footer variable undefined"
    >
      <div
        class="column"
      >
        <h4
          class="h4"
        >
          Sobre nós
        </h4>
        <a
          href="/sobre"
        >
          <p
            class="textoFooter"
          >
            Nossa Cafeteria
          </p>
        </a>
        <a
          href="/sobre/inspiracao"
        >
          <p
            class="textoFooter"
          >
            Nossa Inspiração
          </p>
        </a>
        <a
          href="/sobre/meioAmbiente"
        >
          <p
            class="textoFooter"
          >
            Meio-Ambiente
          </p>
        </a>
        <a
          href="/sobre/contato"
        >
          <p
            class="textoFooter"
          >
            Contate-nos
          </p>
        </a>
      </div>
      <div
        class="column"
      >
        <h4
          class="h4"
        >
          Produtos
        </h4>
        <p
          class="textoFooter"
        >
          Cadastrar produtos
        </p>
        <p
          class="textoFooter"
        >
          Alterar Produtos
        </p>
        <p
          class="textoFooter"
        >
          Visualizar Produtos
        </p>
      </div>
      <div
        class="column"
      >
        <h4
          class="h4"
        >
          Usuarios
        </h4>
        <p
          class="textoFooter"
        >
          Cadastre-se
        </p>
        <p
          class="textoFooter"
        >
          Login
        </p>
        <p
          class="textoFooter"
        >
          Carrinho
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`outros componentes cardItens 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card variable"
      >
        <img
          alt=""
          class="etiqueta"
          data-nimg="1"
          decoding="async"
          height="5000"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fetiqueta.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fetiqueta.png&w=3840&q=75 1x"
          style="color: transparent;"
          width="5000"
        />
        <div
          class="desconto"
        >
          <p
            class="txtDesconto"
          >
            12
            %
          </p>
        </div>
        <div
          class="border"
          style="margin-top: -36%;"
        >
          <img
            alt="nome"
            class="images"
            data-nimg="1"
            decoding="async"
            height="350"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Faffogato.png&w=750&q=75"
            srcset="/_next/image?url=%2Fimages%2Faffogato.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Faffogato.png&w=750&q=75 2x"
            style="color: transparent;"
            width="350"
          />
        </div>
        <h5
          class="nomeProdutos"
        >
          nome
        </h5>
        <p
          class="original"
          style="font-family: fontFamily;"
        >
          123
        </p>
        <p
          class="preco"
          style="font-family: fontFamily;"
        >
          108.24
        </p>
        <a
          href="/produtos/0"
        >
          <button
            class="btn btn-primary botaoComprar"
            type="button"
          >
            Comprar
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card variable"
    >
      <img
        alt=""
        class="etiqueta"
        data-nimg="1"
        decoding="async"
        height="5000"
        loading="lazy"
        src="/_next/image?url=%2Fimages%2Fetiqueta.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Fetiqueta.png&w=3840&q=75 1x"
        style="color: transparent;"
        width="5000"
      />
      <div
        class="desconto"
      >
        <p
          class="txtDesconto"
        >
          12
          %
        </p>
      </div>
      <div
        class="border"
        style="margin-top: -36%;"
      >
        <img
          alt="nome"
          class="images"
          data-nimg="1"
          decoding="async"
          height="350"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Faffogato.png&w=750&q=75"
          srcset="/_next/image?url=%2Fimages%2Faffogato.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Faffogato.png&w=750&q=75 2x"
          style="color: transparent;"
          width="350"
        />
      </div>
      <h5
        class="nomeProdutos"
      >
        nome
      </h5>
      <p
        class="original"
        style="font-family: fontFamily;"
      >
        123
      </p>
      <p
        class="preco"
        style="font-family: fontFamily;"
      >
        108.24
      </p>
      <a
        href="/produtos/0"
      >
        <button
          class="btn btn-primary botaoComprar"
          type="button"
        >
          Comprar
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
